sockeye/inference.py

```diff
@@ -942,6 +949,18 @@ def _beam_search(self,
             #       We currently convert source to NumPy and target ids back to NDArray.
             vocab_slice_ids = mx.nd.array(self.restrict_lexicon.get_trg_ids(source.astype("int32").asnumpy()),
                                           ctx=self.context)
+
+            if vocab_slice_ids.shape[0] < self.beam_size + 1:
+                # This fixes an edge case for toy models, where the number of vocab ids from the lexicon is
+                # smaller than the beam size.
+                logger.warning("Padding vocab_slice_ids (%d) with EOS to have at least %d+1 elements to expand",
+                               vocab_slice_ids.shape[0], self.beam_size)
+                n = self.beam_size - vocab_slice_ids.shape[0] + 1
+                vocab_slice_ids = mx.nd.concat(vocab_slice_ids,
+                                               mx.nd.full((n,), val=self.vocab_target[C.EOS_SYMBOL]),
+                                               dim=0)
+                assert vocab_slice_ids.shape[0] == self.beam_size + 1
```

Comment:
I guess this assert was for development and can be removed, right?

Reply:
yes, will remove